community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NodeRelationshipCache.java

```diff
@@ -410,7 +412,7 @@ public int getCount( long nodeId, Direction direction )
         {
             this.base = base;
             assert chunkSize > 0;
-            this.array = arrayFactory.newDynamicByteArray( chunkSize,
+            this.array = arrayFactory.newDynamicByteArray( 1_000_000,
```

Comment:
oh right, I can revert that.

The thing about calculating number of dense nodes in advance is that it'd be a worst-case figure. We import relationships per type and so there will be a fraction of the entire relationship count for each type, right? The number of relationship groups needed in cache at most is the highest number of dense nodes within one type. We could calculate the biggest group, but that would be like saying that all nodes would be dense in that group.  Perhaps we could calculate the biggest number of dense nodes as well... hmm, lemme get back to you on that.